### PR TITLE
Use public repo as submodule

### PR DESCRIPTION
**Pull request by matthewwalsh0**
# Overview

Use the public `metamask-extension` repository as the extension submodule.

# Changes

## Root

- Update package scripts to use new extension scripts from app workspace.

## Pipeline

- Create job to update the extension Browserify LavaMoat policies for the desktop build type to dynamically add the `external` policies needed for the linked `@metamask/desktop` package.
- Update Chrome version.

## App

- Replace UI components removed from the extension.
- Add package scripts to build all variations of desktop extension.
- Click experimental confirmation button after navigation in extension E2E tests.
- Update extension E2E hooks to resolve failing tests.
- Click risk button and whats new close button in app E2E tests.

### CODE REVIEW

**Comment by matthewwalsh0** [comment]
@SocketSecurity ignore chromedriver@110.0.0 fsevents@1.2.13 playwright@1.23.1

File: packages/app/package.json [hunk 1]
```diff
@@ -11,12 +11,17 @@
...
     "build:app:base": "./build/build.sh",
+    "build:extension": "yarn extension start --build-type desktop",
+    "build:extension:lavamoat": "yarn extension build:dev dev --build-type desktop --apply-lavamoat=true",
+    "build:extension:mv3": "ENABLE_MV3=true yarn build:extension",
+    "build:extension:test": "SKIP_OTP_PAIRING_FLOW=1 yarn extension build:test --build-type desktop",
+    "build:extension:test:app": "yarn extension build:dev dist --apply-lavamoat=true --build-type desktop",
     "build:ui": "node build/ui/desktop-ui.js dev --apply-lavamoat=false",
     "build:ui:ci": "node build/ui/desktop-ui.js dist --apply-lavamoat=false",
-    "build:ui:lavamoat": "node build/ui/desktop-ui.js dev --apply-lavamoat=true",
+    "build:ui:lavamoat": "node build/ui/desktop-ui.js dev --apply-lavamoat=false",
```
Comment (bergarces) [inline packages/app/package.json:24]: Is this intentional?

Comment (matthewwalsh0) [reply packages/app/package.json]: Afraid not, good spot. I initially got everything working with LavaMoat disabled but forgot to revert this one.
Enabled again now.

File: packages/app/src/app/globals.ts [hunk 2]
```diff
@@ -53,7 +53,9 @@ if (!global.self) {
     location: {
       href: 'test.com',
     },
+    /// fetch-with-timeout
     fetch,
+    AbortController,
```
Comment (bergarces) [inline packages/app/src/app/globals.ts:58]: I can't see this defined anywhere.

File: packages/app/src/ui/pages/error/error.component.js [hunk 3]
```diff
@@ -1,35 +1,9 @@
...
-      {renderDesktopError({
-        type: errorType || errorTypeProp,
-        t,
-        isHtmlError: false,
-        history,
-      })}
-    </div>
-  );
+const ErrorPage = () => {
+  return <div className="mmd-error-page"></div>;
```
Comment (bergarces) [inline packages/app/src/ui/pages/error/error.component.js:4]: Is this one intentional? I'm not sure what this component was doing, but wouldn't this always leave it empty?

File: packages/app/src/app/globals.ts (hunk 2 above)
Comment (matthewwalsh0) [reply packages/app/src/app/globals.ts:58]: This is in the global namespace in Node but referenced in `fetch-with-timeout` as `window.AbortController`.

File: packages/app/src/ui/pages/error/error.component.js (hunk 3 above)
Comment (cryptotavares) [reply packages/app/src/ui/pages/error/error.component.js:4]: humm this is the component used on all desktop error pages (exception being the critical error).. I believe that this should be reverted. @matthewwalsh0 was there any specific reason for this change? 🤔

Comment (matthewwalsh0) [reply packages/app/src/ui/pages/error/error.component.js]: I created this branch before there was any of our UI code in the extension so I had to remove the reference here to get the UI building so I could verify the pipeline. Restored.

**Review by cryptotavares — APPROVED** [review]
Looks good!
The only thing that we need to be aware is that we are not splitting running the pipeline between dev CI and a release CI. We have it split just for Extension releases. This means that whenever we are triggering a release of the Desktop App, we are running the e2e tests with an Extension code base that does not match the Extension repo (as it will have a different common package and different lavamoat policies)

**Pull request merged**